hack/test/spider/main.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+var shaA, shaB, commitRange, filterDir string
+
+func init() {
+	flag.StringVar(&shaA, "shaA", "", "First commit in diff calculation")
+	flag.StringVar(&shaB, "shaB", "", "Second commit in diff calculation")
+	flag.StringVar(&commitRange, "commit-range", "", "Range of commits, e.g. shaA...shaB")
+	flag.StringVar(&filterDir, "filter-dir", "", "Directory to filter on")
+
+	flag.Parse()
+}
+
+// canonical takes a full package path and returns the canonical name for it, relative to the repo root.
+func canonical(pkg string) string {
+	return strings.SplitAfter(pkg, "projectcalico/calico/")[1]
+}
+
+// isLocalDir takes a package path and returns whether or not it is a part of the monorepo.
+func isLocalDir(pkg string) bool {
+	return strings.Contains(pkg, "github.com/projectcalico/calico")
+}
+
+func filter(pkg string) string {
+	// Ensure the filterDir ends with a slash.
+	if !strings.HasSuffix(filterDir, "/") {
+		filterDir = filterDir + "/"
+	}
+	if filterDir == "" || strings.HasPrefix(pkg, filterDir) {
```

Comment:
`filterDir != "" && strings.Has....` ?

Reply:
Nope, we want to include it either:
- if there was no filter dir, or
- if the filter dir was given and matches the path.